VCVio/OracleComp/SimSemantics/Simulate.lean

```diff
@@ -20,17 +20,28 @@ For example `idOracle` has no effect upon simulation, and we should apply that f
 
 open OracleSpec Prod
 
-universe u v w
+universe u u' v v' w w' z
 
-variable {ι : Type u} {ιₜ : Type v} {spec : OracleSpec ι} {specₜ : OracleSpec ιₜ}
-  {α β σ : Type w}
+variable {ι ιₜ : Type*} {spec : OracleSpec ι} {specₜ : OracleSpec ιₜ}
+  {α β σ : Type u}
+
+-- T is e.g. `StateT`, `OptionT`, `WriterT`
+structure SimOracle' {ι : Type u} {ιₜ : Type u'}
+    (spec : OracleSpec.{u,v} ι) (specₜ : OracleSpec.{u',v'} ιₜ)
+    (T : (Type w' → Type (max u' (v' + 1) w')) → Type v → Type w) where
+  impl {α : Type v} (q : OracleQuery spec α) :
+    T (OracleComp.{u',v',w'} specₜ) α
+
```

Comment:
`QueryImpl` as an even more general version, allowing an arbitrary monad `m` in the output type